File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.ts
function maxScoreSightseeingPair(values: number[]): number {
    let iMax = values[0];
    let ans = 0;
    for(let i = 1; i < values.length; ++i) {
        ans = Math.max(ans, iMax + values[i] - i);
        iMax = Math.max(iMax, values[i] + i);
    }
    return ans;
};

File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int iMax = values[0]; int ans = 0;
        for(int i = 1; i < values.length; ++i) {
            ans = Math.max(ans, iMax + values[i] - i);
            iMax = Math.max(iMax, values[i] + i);
        }
        return ans;
    }
}